### ISSUE TITLE
stepfunctions - Directly attach a task to a state machine instead of an additional Task class

### ISSUE THREAD

**Issue by nija-at** [aws/aws-cdk#6715]
Currently a task is specified something like -

```
const submitJob = new sfn.Task(this, 'Submit Job', { ... });
const getStatus = new sfn.Task(this, 'Get Job Status', {
    task: new tasks.InvokeFunction(getStatusLambda),
    inputPath: '$.guid',
    resultPath: '$.status',
});
new sfn.StateMachine(this, 'StateMachine', {
    definition: submitJob.next(getStatus),
});
```

Change this to avoid the intermediate `sfn.Task` class, as so -

```ts
const getStatus = new tasks.InvokeFunction((this, 'Get Job Status', {
    handler: getStatusLambda,
    inputPath: '$.guid',
    resultPath: '$.status',
});
```

A case for this -

There is specific guidance on how to [configure retries when using the `InvokeFunction` task](https://docs.aws.amazon.com/step-functions/latest/dg/bp-lambda-serviceexception.html). In the current design, these have to be configured on `sfn.Task`, which is wraps the `InvokeFunction` task, thereby preventing `InvokeFunction` to easily 'vend' its retry best practice and options. If it were to be implemented in the current design, this would require some kind of an additional 'bind' operation, not a great experience for the *stepfunction task authors*.
If this issue's suggestion was implemented, this experience would potentially be slick.

---

This is a :rocket: Feature Request

**Comment by wong-a** [comment]
+1. On my first attempt to create a state machine with a Lambda task in CDK, I was expecting `tasks.RunLambdaTask` to return a Task state that I could chain into my state machine. 

One consideration for the current implementation is that it makes some of the configuration reusable for different states. Consider the example below where the same Lambda function is called in two states that write the output to separate entries in the state output:

```javascript
const echoFn = new lambda.Function(this, 'echoFn', {
  runtime: lambda.Runtime.NODEJS_12_X,
  code: lambda.Code.fromInline(
    `exports.handler = async (event) => ({ statusCode: 200, body: event })`
  ),
  handler: 'index.handler',
})

const taskDefinition = new tasks.RunLambdaTask(echoFn);

const lambdaTask1 = new sfn.Task(this, 'CallLambda1', {
  task: taskDefinition,
  resultPath: '$.task1Result'
})
const lambdaTask2 = new sfn.Task(this, 'CallLambda2', {
  task: taskDefinition,
  resultPath: '$.task2Result'
})
```

**Comment by shivlaks** [comment]
closed by #7946 - tracking for the rest of the service integrations can be found at #6489
